### PR TITLE
Split and remove pkg/gitutils and pkg/httputils

### PR DESCRIPTION
**Pull request by dnephin**
See #32989

Moves `pkg/gitutils` to `builder/remotecontext/git` where it is used.
Split up `pkg/httputils`. Remove unused functions, move the rest to the single consumer. There is one function left `Download()`, which is very small. It can be duplicated when components are split out.

### CODE REVIEW

**Comment by vieux** [comment]
LGTM ping @cpuguy83

**Comment by vieux** [comment]
ping @aaronlehmann

**Comment by cpuguy83** [comment]
Do we need an `httpdownload` package?

**Comment by cpuguy83** [comment]
Actually definitely no, `httpdownload.Download` is pretty worthless.

**Comment by dnephin** [comment]
Right. It's a minor piece of code. When we split things we can just duplicate it. I don't see any value in duplicating it now.

**Comment by cpuguy83** [comment]
@dnephin Value is not creating a new package, and there's nothing really to copy.

**Comment by cpuguy83** [comment]
By "there's nothing really to copy" I mean... the function is really that worthless. It's `http.Get` with an error check that masks the real error.

**Comment by dnephin** [comment]
Ya, it probably should have included the response body instead of just the status code.

File: builder/remotecontext/git.go [hunk 1]
```diff
@@ -4,13 +4,13 @@ import (
 	"os"
 
 	"github.com/docker/docker/builder"
+	"github.com/docker/docker/builder/remotecontext/git"
 	"github.com/docker/docker/pkg/archive"
-	"github.com/docker/docker/pkg/gitutils"
```
Comment (vdemeester) [inline builder/remotecontext/git.go]: I know this package is used in several projects outside (like `libcompose`, …).

Comment (dnephin) [inline builder/remotecontext/git.go]: They will have to change their import paths to `build/remotecontext/git`.

**Comment by dnephin** [comment]
I removed `pkg/httpdownload`, and I fixed the function to do what it should have been doing.

**Review by cpuguy83 — APPROVED** [review]
LGTM

**Comment by vieux** [comment]
LGTM

**Pull request merged**